bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/nhc2/NikoHomeControlCommunication2.java

```diff
@@ -549,7 +540,8 @@ private void updateThermostatState(NhcThermostat2 thermostat, NhcDevice2 device)
         Optional<NhcProperty> ambientTemperatureProperty = device.properties.stream()
                 .filter(p -> (p.ambientTemperature != null)).findFirst();
```

Comment:
```suggestion
        Optional<Integer> ambientTemperature = device.properties.stream().map(p -> p.ambientTemperature).filter(Objects::nonNull).map(t -> Math.round(Float.parseFloat(t) * 10)).findFirst();      
```

and later

```
        int measured = ambientTemperatureProperty.orElse(thermostat.getMeasured());
```

Reply:
Thanks.

Reply:
I also changed a few of the other assignments in here.